sdk/tests/iot/common/CMakeLists.txt

```diff
@@ -18,5 +18,4 @@ add_cmocka_test(az_iot_common_test SOURCES
                 LINK_TARGETS
                     az_iot_common
                     az_core
-                    ${PAL}
```

Comment:
I believed iot were using clock, at some point..  calling wait(), or something like that. Linker was failing to find the function definition if not linking a PAL.  Did that change?

Reply:
IoT does not have a dependency on PAL and we're not/should not be using using clock there. Precondition no longer invokes sleep for exactly this reason.
It all builds, links, runs, and passes, so no problem.

Reply:
nit: In the future, please do such cleanup changes in a separate PR/commit.